### PR TITLE
6 add actions gitignore and docs

### LINKED ISSUE #6: Add actions, .gitignore and docs

### PR DESCRIPTION
**Pull request by Pepe-Marquez**
In this PR I have:

- Modified the gitignore to be more complete and avoid pushing some unwanted artifacts 
- Added pipelines according to the standard NOMAD plugins (I have commented the ruff ones)
- Added a basic doc structure for us to log how-tos and tutorials and have basic explanation that we can deploy.

### CODE REVIEW

**Comment by Pepe-Marquez** [comment]
I am struggling to pass the pipelines because of some dependencies coming from the baseclasses plugin. I wonder if we could include them in the pyproject.toml there and if this wold solve the issues.

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: install-and-test-workflow
```
Comment (RoteKekse) [inline .github/workflows/actions.yml:1]: what is this doing?

File: .github/workflows/cruft-update.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# /.github/workflows/cruft-update.yml
```
Comment (RoteKekse) [inline .github/workflows/cruft-update.yml]: what is this doing?

File: docs/reference/references.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# References
```
Comment (RoteKekse) [inline docs/reference/references.md:1]: is this needed here?

File: pyproject.toml [hunk 4]
```diff
@@ -28,9 +28,18 @@ maintainers = [
     { name = "Michael Götte", email = "michael.goette@helmholtz-berlin.de" },
 ]
 license = { file = "LICENSE" }
-#dependencies = ["nomad-lab>=1.2.2dev578"]
+dependencies = [
+    "nomad-lab>=1.2.2dev578",
```
Comment (RoteKekse) [inline pyproject.toml:33]: i think this is the same issue as in baseclass plugin and might not be working?

**Review by RoteKekse — COMMENTED** [review]
hey @Pepe-Marquez i need to make sure that i understand all of this to some degree before deploying it into production

File: .github/workflows/actions.yml (hunk 1 above)
Comment (blueraft) [reply .github/workflows/actions.yml:1]: Runs the tests, it also generates some coverage data.

File: .github/workflows/cruft-update.yml (hunk 2 above)
Comment (blueraft) [reply .github/workflows/cruft-update.yml]: This is not needed since this project wasn't generated by cruft

File: pyproject.toml (hunk 4 above)
Comment (blueraft) [reply pyproject.toml:33]: The issue here is newer versions of `nomad-lab` were only released on our Gitlab pypi server. If you want to install it, you'll have to pass the extra index url. This is mentioned in the [readme](https://github.com/FAIRmat-NFDI/cookiecutter-nomad-plugin/tree/main/nomad-%7B%7Bcookiecutter.plugin_name%7D%7D#install) of cookiecutter plugins.

You can install it the following way: 
```sh
pip install -e '.[dev]' --index-url https://gitlab.mpcdf.mpg.de/api/v4/projects/2187/packages/pypi/simple
```

Comment (RoteKekse) [reply pyproject.toml:33]: Can this be added to the Pytoml file?

File: .github/workflows/cruft-update.yml (hunk 2 above)
Comment (RoteKekse) [reply .github/workflows/cruft-update.yml]: Ok will remove it then

File: .github/workflows/actions.yml (hunk 1 above)
Comment (RoteKekse) [reply .github/workflows/actions.yml:1]: Ok will comment until there are tests

Comment (blueraft) [reply .github/workflows/actions.yml:1]: https://github.com/nomad-hzb/hzb-combinatorial-libraries-plugin/blob/6-add-actions-gitignore-and-docs/tests/test_schema.py there's one test

File: pyproject.toml (hunk 4 above)
Comment (blueraft) [reply pyproject.toml:33]: pip doesn't support that.

You could use [uv](https://github.com/astral-sh/uv) instead, which supports this feature.

```toml
[tool.uv.pip]
index-url = "https://gitlab.mpcdf.mpg.de/api/v4/projects/2187/packages/pypi/simple"
```

`uv pip install -e '.[dev]'` will work then.

**Pull request merged**